### ISSUE TITLE
Gettin' piggy with it

### ISSUE THREAD

**Issue by jamesdabbs** [TIY-DC-ROR-2015-Jan/course-notes#59]
_Due on Feb 02 @ 09:00am_
## Learning Objectives
- Modeling a real-world game, and decomposing your process into small pieces (methods)
- More practice with variable scope and passing, `Array`s and `each`
### Normal Mode

Try to implement all of these without referring to the example from class:
- Reimplement the game, but with the "Hog" variant: each turn, the game asks the player how many dice to roll. If any of them are 1, they get no points; otherwise they get the total.
- Ask the players for their names, and print those out at each turn.
- When the game is over, ask if the player(s) want to play again.
- Present a better "divider" between turns - could be cool ASCII art, or a 2 second pause, or waiting for a player to press enter, or ...
- Allow any number of players >2

Once that's done, refer to (/copy) the implementation from class and add:
- Let the player decide which variant (Hog or Pig) they want to play.
### Hard Mode

Add the following extra features
- Let the player ask for a hint showing their _expected_ score if they roll again. (Question: is taking the hint always the best play?)
- Build an AI to play against the player. Let the player pick 1- or 2- player mode before starting the game.
- Add a two-dice variant - players roll two dice at at time, lose _all_ of their points on double-1s, and must roll again after any other doubles.

**Comment by kwals** [comment]
https://github.com/kwals/files/blob/master/play_hog

I tried to figure out a class for player but couldn't get it to work. Will try to rewrite after we go over classes in class.

**Comment by jamesdabbs** [comment]
Looks good by-and-large. We'll be refining program structure over the week, so keep it in the back of your head and think about how you could improve this one. A few minor suggestions:
- It'd be nice if it said "What is player #{n}'s name" instead of "What is your name" multiple times.
- Is [roll_die](https://github.com/kwals/files/blob/master/play_hog#L5) necessary as written? Isn't it just equivalent to `rand(1..6)` since you're always returning after doing it once?
- Your indentation still looks wonky (which _might_ be because Sublime doesn't recognize this as a ruby file because it doesn't end in .rb). I don't much care about 2-spaces or 4-spaces or whatever, but stuff like [indenting after a new block](https://github.com/kwals/files/blob/master/play_hog#L66) is important for readability.

Nice work! :+1:

**Comment by kwals** [comment]
I'm having the same problem with indentation... I thought I fixed in in Sublime, but when I pust to GitHub, it ends up having a lot of extra spaces. What's the folder shortcut thing you showed us in lecture again?

**Comment by jamesdabbs** [comment]
This is a good write-up: http://stackoverflow.com/questions/9474090/how-do-i-force-sublime-text-2-to-indent-two-spaces-per-tab

That probably won't fix old files that are already weirdly indented. You may need to either select all and force a re-indent, or click on tab settings at the bottom right and select "convert tabs to spaces".

**Comment by kwals** [comment]
Sorry, didn't realize I hadn't closed this.
